### PR TITLE
Part 1: Initial GUI with user input

### PR DESCRIPTION
**Pull request by KuchtaVR6**

### CODE REVIEW

File: src/components/MainPage.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const MainPage: React.FC = () => {
+
+    return (
+        <>
+            <div className = "user-area-form-container">
+                <UserInputFormArea />
+            </div>
+            <div className = "train-board-container">
+                <TrainBoard />
+            </div>
```
Comment (emilies03) [inline src/components/MainPage.tsx]: Should this div and styling be within the component?

if user-area-form-container and train-board-container are always needed for these components you will make the user's life easier by moving them inside

File: src/components/StationSelectMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                
+                <option label = " " style = { { display: 'none' } } selected = { true }></option>
+
+                {stationList.map((stationEntry) => {
+                    if (!skipTheseStationIDs.includes(stationEntry.codes.id)) {
+                        return (
+                            <option 
+                                value = { stationEntry.codes.id } 
+                                key = { stationEntry.codes.id }
+                                onClick = { () => {setSelection(stationEntry);} }>
```
Comment (emilies03) [inline src/components/StationSelectMenu.tsx]: spoken about in person but tabbing will not trigger this change

File: src/components/UserInputFormArea.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                <StationSelectMenu 
+                    label = 'Destination:' 
+                    stationList = { hardCodedStations } 
+                    setSelection = { setDestinationStation }
+                    skipTheseStationIDs = { [departureStation?.codes.id] }
+                />
+            </div>
+            
+            <div className = 'station-submit-area'>
+                <div className = 'warning-message'>{errorMessage}</div>
```
Comment (emilies03) [inline src/components/UserInputFormArea.tsx]: should an error div always exist even when there is no error?

**Review by emilies03 — COMMENTED** [review]
This looks great well done! The naming and structuring is very clear.

I've just left a couple of comments to think about :)

File: src/components/MainPage.tsx (hunk 1 above)
Comment (KuchtaVR6) [reply src/components/MainPage.tsx]: Yes, it's now refactored.

File: src/components/StationSelectMenu.tsx (hunk 2 above)
Comment (KuchtaVR6) [reply src/components/StationSelectMenu.tsx]: Fixed, now can be easily navigated with Tab.

File: src/components/UserInputFormArea.tsx (hunk 3 above)
Comment (KuchtaVR6) [reply src/components/UserInputFormArea.tsx]: Removing this div would mean an separate set of styling rules for the case where there is no error, as such we kept the div but renamed it to minimize the risk of confusion.

File: src/components/UserInputFormArea.tsx [hunk 4]
```diff
@@ -55,11 +56,11 @@ const hardCodedStations : StationInfo[] = [
     },
 ];
 
-const UserInputFormArea : React.FC = () => {
+const UserInputFormArea: React.FC = () => {
 
     const [departureStation, setDepartureStation] = useState<StationInfo>();
     const [destinationStation, setDestinationStation] = useState<StationInfo>();
-    const [errorMessage, setErrorMessage] = useState('Please select both stations');
+    const [message, setErrorMessage] = useState('Please select both stations');
```
Comment (emilies03) [inline src/components/UserInputFormArea.tsx]: The set method should match the name of the state :)

**Review by emilies03 — APPROVED** [review]
Looks great, just 1 naming issue. But you can merge in once that is fixed!

**Pull request merged**